### PR TITLE
BUG: Fix parameter validity checks in ``random.choice``.

### LINKED ISSUE #13087: np.random.choice does not raise the exception when size negative and replace=False
np.random.choice does not raise the exception when size negative and replace=False

### Reproducing code example:
```python
import numpy as np
print(np.random.choice(range(2000000), -85, replace=False).shape)
```
return (1999915,)

```python
import numpy as np
print(np.random.choice(range(2000000), -85).shape)
```
raise ValueError:
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "mtrand.pyx", line 1162, in mtrand.RandomState.choice
  File "mtrand.pyx", line 994, in mtrand.RandomState.randint
  File "mtrand.pyx", line 995, in mtrand.RandomState.randint
  File "randint_helpers.pxi", line 253, in mtrand._rand_int64
ValueError: negative dimensions are not allowed

### Numpy/Python version information:
1.16.0 3.5.2 (default, Nov 23 2017, 16:37:01) 
[GCC 5.4.0 20160609]

### PR DESCRIPTION
**Pull request by bharatr21**
Fix #13087 by raising a ValueError with the same error message and adding a unit test for it

### CODE REVIEW

File: numpy/random/tests/test_random.py [hunk 1]
```diff
@@ -400,6 +400,7 @@ def test_choice_exceptions(self):
         assert_raises(ValueError, sample, [1, 2], 3, p=[1.1, -0.1])
         assert_raises(ValueError, sample, [1, 2], 3, p=[0.4, 0.4])
         assert_raises(ValueError, sample, [1, 2, 3], 4, replace=False)
+        assert_raises(ValueError, sample, [1, 2, 3], -2, replace=False)
```
Comment (eric-wieser) [inline numpy/random/tests/test_random.py]: Can you test with `(-1,)` and `(-1, 1)` too?

Comment (Bharat123rox) [reply numpy/random/tests/test_random.py]: @eric-wieser You mean `size=(-1,1)`? Shouldn't it be a 1D array (is that what we're checking here)?

Comment (eric-wieser) [reply numpy/random/tests/test_random.py]: Yes, I do mean that. The first argument is a 1d array, but the output shape can be any dimension. Your new tests look good, although you might want to reference the issue number as `# gh-NNNNN` in a comment

Comment (Bharat123rox) [reply numpy/random/tests/test_random.py]: Added `# gh-<issueno>`
@eric-wieser Thanks for patiently reviewing this PR! Is there anything more?

**Review by eric-wieser — APPROVED** [review]

**Comment by mattip** [comment]
Thanks @Bharat123rox

**Pull request merged**